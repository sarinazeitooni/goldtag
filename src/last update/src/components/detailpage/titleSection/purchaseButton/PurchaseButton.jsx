import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faShoppingBag} from "@fortawesome/free-solid-svg-icons";
import NumberFormat from "react-number-format";
import React from "react";
function PurchaseButton({id , data}){
    let discountPercent = ((data.price - data.newprice) * 100) / data.price;
    return(
        <button className='price-container-btn'>
            <div className='shopping-icon'><FontAwesomeIcon icon={faShoppingBag} /></div>
            <div>
                <div className='new-price-container'><span>{id.purchase}</span><span className='price-num'>
                       <NumberFormat className='' value={data.newprice} displayType={'text'}
                                     thousandSeparator={true} renderText={(value, props) =>
                           <span {...props}>{value}</span>
                       }/>

                    </span>
                    <span className='currency'>{id.currency}</span></div>
                <div className='old-price-container'><span className='discount-percent'> {discountPercent}{id.discount}</span>
                    <span className='old-price'><NumberFormat className='' value={data.price} displayType={'text'}
                                                              thousandSeparator={true} renderText={(value, props) =>
                        <span {...props}>{value}</span>
                    }/>{id.currency}</span></div>
            </div>
        </button>
    )
};
export default PurchaseButton;