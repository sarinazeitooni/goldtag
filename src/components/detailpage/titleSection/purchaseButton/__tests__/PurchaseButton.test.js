import React from 'react';
import { shallow } from 'enzyme';
import PurchaseButton from "../PurchaseButton";
describe("PurchaseButton component" , ()=>{
    it("it should render a div",()=>{
        const container = shallow(<PurchaseButton data={''} id={''}/>)
        expect(container).toMatchSnapshot();
    });
});