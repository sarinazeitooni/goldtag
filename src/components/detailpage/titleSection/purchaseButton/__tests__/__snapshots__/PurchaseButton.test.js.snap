// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseButton component it should render a div 1`] = `ShallowWrapper {}`;
